packages/controller-utils/src/caip.ts

```diff
@@ -0,0 +1,75 @@
+import type { CaipChainId, Hex } from '@metamask/utils';
+import { parseCaipChainId, isCaipChainId } from '@metamask/utils';
+
+import { toHex } from './util';
+
+/**
+ * Checks whether the given value is a valid CAIP chain ID string for Ethereum.
+ *
+ * @param caipChainId - The CAIP chain ID to check for safety.
+ * @returns Whether the given CAIP chain ID is valid for Ethereum chains.
+ */
+export function isEthCaipChainId(
+  caipChainId: unknown,
+): caipChainId is CaipChainId {
+  if (!isCaipChainId(caipChainId)) {
+    return false;
+  }
+  const { namespace, reference } = parseCaipChainId(caipChainId);
+  const chainId = parseInt(reference, 10);
+  return namespace === 'eip155' && !Number.isNaN(chainId);
+}
+
+/**
+ * Creates a CaipChainId string that specifies an Ethereum chain.
+ *
+ * @param chainId - The chain ID in 0x prefixed hex or decimal string.
+ * @returns a valid CAIP chain ID for an Ethereum chain.
+ */
+export function buildEthCaipChainId(chainId = ''): CaipChainId {
+  const chainIdDecimal = chainId.startsWith('0x')
+    ? parseInt(chainId, 16).toString(10)
+    : chainId;
+
+  if (Number.isNaN(parseInt(chainIdDecimal, 10))) {
+    throw new Error(`Invalid chain ID "${chainId}"`);
+  }
+  return `eip155:${chainIdDecimal}` as CaipChainId;
+}
+
+/**
+ * Parses an Ethereum CAIP chain ID into a decimal string.
+ *
+ * @param caipChainId - The Ethereum CAIP chain ID string.
+ * @returns the chain ID as a decimal string.
+ */
+export function parseEthCaipChainId(caipChainId: CaipChainId): string {
+  const { reference } = parseCaipChainId(caipChainId);
+  return reference;
+}
+
+/**
+ * Parses an Ethereum CAIP chain ID into a hex string.
+ *
+ * @param caipChainId - The Ethereum CAIP chain ID string.
+ * @returns the chain ID as a hex string.
+ */
+export function parseEthCaipChainIdHex(caipChainId: CaipChainId): Hex {
+  const { reference } = parseCaipChainId(caipChainId);
+  return toHex(reference);
+}
+
+/**
+ * Parses an Ethereum CAIP chain ID into an integer.
+ *
+ * @param caipChainId - The Ethereum CAIP chain ID string.
+ * @returns the chain ID as an integer.
+ */
+export function parseEthCaipChainIdInt(caipChainId: CaipChainId): number {
```

Comment:
alternative names for these helpers could be

```
toEthCaipChainId
toEthChainId
toEthChainIdHex
toEthChainIdInt
```

which would be done to match `toHex`

Reply:
Agreed that there could be some better naming on these conversion helpers. 

Second, should we ever support turning a chainId into an actual integer? Im not sure its 100% clear wether max chainId fits safely in an integer. It also might get confusing if some places have a base10 string, some places have a base10 integer... can we consolidate the format into being always a string? maybe thats outside the scope of this PR, just putting it out there though.

Reply:
Regardless of whether or not it is safe, we are already doing this conversion in a few places. From what I've seen, the preference in our usage is Hex -> Decimal (string) -> Decimal (number). We generally only do the conversion from Decimal string to number when we are comparing against chainId number json values returned by some outside API

Reply:
Renamed the helpers